### DISCUSSION TITLE
MODPATH for USGS GSFLOW simulation: problem with budget file

### DISCUSSION THREAD

**Discussion by porterma** [modflowpy/flopy#1582] [Q&A] [answered]
I am trying to run MODPATH6 on a USGS GSFLOW simulation [model](https://water.usgs.gov/GIS/metadata/usgswrd/XML/sir2017-5097.xml)

So my first question, is MODPATH6 compatible with GSFLOW?

I assumed it was so I gave it a try. MODPATH6 crashes with this error in command prompt

> Processing basic data ...
> Checking head file ...
> Checking budget file and building index ...
> forrtl: severe (170): Program Exception - stack overflow
> Image              PC        Routine            Line        Source
> mp6.exe            004B0787  Unknown               Unknown  Unknown
> mp6.exe            00428174  Unknown               Unknown  Unknown
> mp6.exe            00420C79  Unknown               Unknown  Unknown
> mp6.exe            00445E4A  Unknown               Unknown  Unknown
> mp6.exe            004CC213  Unknown               Unknown  Unknown
> mp6.exe            004B0C59  Unknown               Unknown  Unknown
> KERNEL32.DLL       7671B028  Unknown               Unknown  Unknown

The mplist file looks like this towards the bottom.

> THE HEAD FILE FOR GRID   1 IS COMPLETE.
>  
>  CHECKING AND INDEXING BUDGET FILE FOR GRID   1
>  Index position =          1
>  READ FLOW PACKAGE BUDGET DATA FOR STRESS PERIOD    1 , TIME STEP    1 ...
>  Single Precision Binary Budget file
>  Reading          STORAGE for Period    1   Time Step    1
>  Reading    CONSTANT HEAD for Period    1   Time Step    1
>  Reading FLOW RIGHT FACE  for Period    1   Time Step    1
>  Reading FLOW FRONT FACE  for Period    1   Time Step    1
>  Reading FLOW LOWER FACE  for Period    1   Time Step    1
>    FLOW RATES BETWEEN CELLS HAVE BEEN READ
>  Reading   STREAM LEAKAGE for Period    1   Time Step    1
>  Reading     UZF RECHARGE for Period    1   Time Step    1

So it appears stream leakage is read fine, but not recharge for the uzf package in the budget file.

The portion of the nam package for output looks like this.

> data(binary)     50   ../../output/output_nstp/heads.bin
> data(binary)     55   ../../output/output_nstp/bud_cbc.bin
> data(binary)     49   ../../output/output_nstp/bud_uzf.bin
> data             56   ../../output/output_nstp/str_out.txt
> data             41   ../../output/output_nstp/hob_out.txt

My beginning of the oc package looks like this and is saving head and budget data for every timestep and stress period..

> head print format 0
> head save unit 50
> COMPACT BUDGET AUXILIARY
> 
> period 1 step 1
> save head
> save budget
> 
> period 1 step 2
> save head
> save budget

Item 1c of the sfr package looks like this and ISTCB1=55 

> -1969   118   0   0 1.486      0.00001           55           56            1       0

Item 1b of the uzf package looks like this where IUZFCB1=55.

> 3 1 0 0 55 49 5 20 0 1.0   NUZTOP IUZFOPT IRUNFLG IETFLG IUZFCB1 IUZFCB2 NTRAIL2 NSETS2 NUZGAG  SURFDEP

In my well package IWELCB=55.

There is an option in the UZF to set IUZFCB1<0 (e.g.,-55), which saves infiltration and unsaturated ET to binary file. Do I need IUZFCB1=-55? 

There is also IUZFCB2 that writes to unformatted file. I currently have that set to a separate ASCII file. Should this be set to 55 and written as binary as well?

Any thoughts would be greatly appreciated. I will test these ideas myself however the simulation take ~1.5 days, so I thought I would ask here for some guidance.

Also, I apologize if this my second discussion ticket regarding this. I wrote one last night and thought I posted, but do not see it on github.

**Comment by jlarsen-usgs** [comment]
Hi @porterma, 

I'm not sure if MODPATH is compatible with GSFLOW.  @rniswon should be able to provide more insight into this.

**Comment by rniswon** [comment]
I am in communication with Mark. MODPATH is not compatible with GSFLOW. However, results from a GSFLOW model can be exported and converted to a MODFLOW only model with equivalent boundary fluxes, and this model can be used with MODPATH. I am working to connect Mark with someone who has done this in the past.

**Comment by porterma** [comment, marked as answer]
Apologies for the delayed update here. 

I was able to get MODPATH running on the GSFLOW budget file. I simply removed the UZF out put from the budget file and the MODPATH works as expected. I realize I am missing the fluxes to and from the UZF package. However, for my purpose of estimating broad particle travel time ranges this is sufficient.

I was not bale to get MODFLOW to run with the the UZF budget data. I started down a path of running MODFLOW alone with the fluxes from PRMS converted to boundary conditions to MODFLOW. However, I did not spend sufficient time on it to come up with a solution. 

@rniswon I appreciate your assistance on and advice with this task.
